portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro

```diff
@@ -136,6 +136,60 @@ definition {
 			ACSettings.deleteDataSource();
 		}
 	}
+	
+	macro createRequestLogByEmail {
+		var key_fieldType = "${fieldType}";
+
+		ACNavigation.goToSettings();
+
+		ACSettings.goToDataControlAndPrivacy();
+
+		Click(locator1 = "ACDataControlAndPrivacy#RESQUEST_LOG");
+
+		Click(locator1 = "ACDataControlAndPrivacy#CREATE_REQUEST");
+
+		for (var jobType : list "${jobTypes}") {
+			Click(
+				locator1 = "ACDataControlAndPrivacy#JOB_TYPE",
+				key_jobType = "${jobType}");
+		}
+
+		Click(
+			locator1 = "ACDataControlAndPrivacy#DATA_ID",
+			key_fieldType = "Find by Email");
+
+		Click(locator1 = "ACDataControlAndPrivacy#FIND_BY_EMAIL_FIELD");
+
+		for (var emailAdress : list "${emailAdresses}") {
+			Click(
+				locator1 = "ACDataControlAndPrivacy#FIND_EMAIL_FIELD",
+				key_emailAdress = "${emailAdress}");
+		}
+
+		Button.clickSave();
+
+		Alert.viewSuccessMessage();
+
+		for (var emailAdress : list "${emailAdresses}",) {
+			if ("${jobType}" == "Delete") {
+				AssertElementPresent(
+					locator1 = "ACDataControlAndPrivacy#REQUEST_LOG_TABLE",
+					key_emailAdress = "${emailAdress}",
+					key_requestType = "${jobType}");
+					
+				AssertElementPresent(
+					locator1 = "ACDataControlAndPrivacy#REQUEST_LOG_TABLE",
+					key_emailAdress = "${emailAdress}",
+					key_requestType = "Suppress");
+			}
+			else {
+				AssertElementPresent(
+					locator1 = "ACDataControlAndPrivacy#REQUEST_LOG_TABLE",
+					key_emailAdress = "${emailAdress}",
+					key_requestType = "${jobType}");
+			}
+		}
```

Comment:
I think you'd better put the assert macro separate from the create macro. So you can have the option to assert without having to create the request.

Reply:
I think you forgot the assert option if the person only mark the Suppress option

Reply:
fixed